gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/converters/EntityServiceAndGatewayServiceConverter.java

```diff
@@ -205,6 +197,34 @@ private static Function convertToQueryFunction(
         {
           builder.setFunctionName(function.getFunction().name()).setAlias(function.getAlias());
 
+          // Backward compatibility to handle long values
+          if (function.getFunction() == FunctionType.AVGRATE) {
```

Comment:
Actually, now that I look at the next file - isn't this the converter for Entity Service? We didn't add this support to ES that I saw, so this file can remain unmodified (comments here generally apply in the other converter though). That said, not sure that we ever use or would want to use avg_rate against ES - it has no concept of time, so maybe it would make sense to remove this entirely (not a bad idea considering we've changed the validator and it no longer aligns with the assumptions here).

Reply:
> Actually, now that I look at the next file - isn't this the converter for Entity Service? We didn't add this support to ES that I saw, so this file can remain unmodified (comments here generally apply in the other converter though). That said, not sure that we ever use or would want to use avg_rate against ES - it has no concept of time, so maybe it would make sense to remove this entirely (not a bad idea considering we've changed the validator and it no longer aligns with the assumptions here).

@kotharironak Aaron's comment which got deleted due to some ui issue on git.